# QC and Overview

    1. Loading both Parquet files

    2. Quick QC: number of rows/columns, missing values, duplicates

    3. Column overview: listing the main columns and what they represent

In [1]:
import pandas as pd
import numpy as np

# Load both Parquet Files

In [2]:
mapped = pd.read_parquet("data_sampled_10_percent.parquet")
metadata = pd.read_parquet("lookup.parquet")

- Mapped peptide file contains the peptide mapping results. 
- Metadata file serves as a lookup table for annotations.

# Quick QC: Number of Rows/Columns, Missing Values, Duplicates

#### Number of rows & columns

In [3]:
mapped.shape

(97744896, 4)

In [4]:
metadata.shape

(4353709, 9)

- Mapped peptide file contained 97,744,896 rows and 4 columns. 
- Metadata file contained 4,353,709 rows and 9 columns. 

#### Missing values

In [6]:
mapped.isna().sum()

peptide      0
side         0
accession    0
position     0
dtype: int64

- Mapped peptide file didn't contain any missing values and duplicated rows.

In [7]:
metadata.isna().sum()

accession          0
length             0
taxid              0
title              0
species         7260
strain       4353709
lineage_x    4353709
family       1201843
lineage_y     398842
dtype: int64

- Metadata file contains missing values for the species, strain, lineage_x, family, lineage_y columns. strain, lineage_x columns are fully missing.
-  Because of the great diversity among viruses, it is hard to classify them. This may be the reson of the missing values in strain column.

#### Count duplicates

In [8]:
mapped.duplicated().sum()

np.int64(0)

In [41]:
metadata.duplicated().sum()

- No completely duplicated rows were detected in either the mapped peptides table or the metadata table. However, there may be multiple similarities in rows within the mapped data. This is due to the characteristics of the data itself. 

# QC Summary Table

In [11]:
def QC(df,second_col):
    shape = df.shape

    missing_val = df.isna().sum()

    is_duplicated = df.duplicated().sum()

    return print(f"Number of rows and columns: {shape}\nMissing_values: {missing_val}\nDuplicates: {is_duplicated}\n")
    

In [15]:
# mapped QC

print(QC(mapped, "position"))

Number of rows/columns: (97744896, 5)
Missing_values: peptide       0
side          0
accession     0
position      0
duplicates    0
dtype: int64
Duplicates: 0

None


In [20]:
# metadata QC

print(QC(metadata, "taxid"))

Number of rows/columns: (4353709, 10)
Missing_values: accession           0
length              0
taxid               0
title               0
species          7260
strain        4353709
lineage_x     4353709
family        1201843
lineage_y      398842
duplicates          0
dtype: int64
Duplicates: 0

None


# Column Overview

### 1. Mapped peptide

In [15]:
mapped.columns

Index(['peptide', 'side', 'accession', 'position'], dtype='object')

- peptide: Amino acid sequence.
- side: Represents whether the peptide belongs to a virus or a human.
- accesion: Unique identifiers for proteins assigned by the NCBI database.
- position: Starting position of the peptide within the protein.

### 2. Metadata

In [16]:
metadata.columns

Index(['accession', 'length', 'taxid', 'title', 'species', 'strain',
       'lineage_x', 'family', 'lineage_y'],
      dtype='object')

- accession: Unique identifiers for proteins assigned by the NCBI database.
- length: Length of the peptide chain. 
- taxid: Unique identifier assigned by the NCBI to the source organism of the protein. 
- title: Name of the protein or ORFs with the description of organism.
- species: Species name related with the protein.
- strain: Strain is a genetic variant, a subtype within a biological species.
- lineage_x: Taxonomic lineage of the species. 
- family: Family annotations of viruses. 
- lineage_y: Taxonomic lineage of the species.